logstash-core/lib/logstash/java_pipeline.rb

```diff
@@ -255,6 +259,19 @@ def start_workers
     end
   end
 
+  def resolve_cluster_uuids
```

Comment:
alternate implementation:
 - relies on `Set` to eliminate some conditional logic
 - uses `Enumerable#each_with_object` to reduce scope-jumping:
 - composes the setting key once, using string interpolation which is safer (avoiding sending `Object#+(String)`, which can behave or fail in surprising ways if the receiver is not a String)

~~~ ruby
  def resolve_cluster_uuids
    outputs.each_with_object(Set.new) do |output, cluster_uuids|
      cluster_uuid_setting_key = "#{output.id}.cluster_uuid"
      if LogStash::Settings.registered?(cluster_uuid_setting_key)
        cluster_uuids << LogStash::SETTINGS.get(cluster_uuid_setting_key)
      end
    end.to_a
  end
~~~